tests/unit/lms/services/canvas_test.py

```diff
@@ -0,0 +1,45 @@
+from unittest.mock import sentinel
+
+import pytest
+
+from lms.services import CanvasFileNotFoundInCourse, CanvasService
+from lms.services.canvas import factory
+
+
+class TestCanvasService:
+    @pytest.mark.parametrize("check_in_course", (True, False))
+    def test_public_url_for_file(self, canvas_service, check_in_course):
+        canvas_service.api.list_files.return_value = [{"id": sentinel.file_id}]
+
+        result = canvas_service.public_url_for_file(
+            file_id=sentinel.file_id, check_in_course=check_in_course, course_id="*any*"
+        )
+
+        assert result == canvas_service.api.public_url.return_value
+        canvas_service.api.public_url.assert_called_once_with(sentinel.file_id)
+
+    def test_public_url_for_file_with_unsuccessful_file_check(self, canvas_service):
+        canvas_service.api.list_files.return_value = []
+
+        with pytest.raises(CanvasFileNotFoundInCourse):
+            canvas_service.public_url_for_file(
+                file_id=sentinel.file_id,
+                course_id=sentinel.course_id,
+                check_in_course=True,
+            )
```

Comment:
I think you might want to add a test that `CanvasService` raises if `CanvasAPIClient.public_url()` or `list_files()` raises

Reply:
Is there a specific scenario you had in mind? Presumably any code calling any other code and not catching will raise. I don't like testing the negative here (we don't have any code that...) as there's no real limit to it.

Reply:
I think there's a finite number of particularly important exceptions that we raise and that get handled by exception views in particular ways, e.g. I think `CanvasAPIClient` raises `CanvasAPIError` or something. It's a pretty finite number: our call stack tends not to be very deep. A view calling a service that calls another service is about as deep as it usually gets.

On other hand perhaps the exception raise just needs to be documented on the one service method that directly raises the exception and not repeated on other service and view methods that call it, and that will be enough to assist developers in ensuring that the exception's always gonna be caught by an exception view that's gonna handle it appropriately